apps/communications/contacts/views/details/js/boot.js

```diff
@@ -45,6 +45,55 @@ window.onload = function() {
       document.body.classList.remove('hidden');
     });
 
+
+    function getParams() {
+      var params = {};
+      var raw = window.location.search.split('?')[1];
+      var pairs = raw.split('&');
+      for (var i = 0; i < pairs.length; i++) {
+
+        var data = pairs[i].split('=');
+        params[data[0]] = data[1];
+      }
+      return params;
+    }
+
+    function renderContact(id) {
+      return new Promise(function(resolve, reject) {
+        ContactsService.get(params.contact, function onSuccess(savedContact) {
+          ContactsService.getCount(count => {
+            resolve(savedContact, count);
+          });
+        }, function onError() {
+          console.error('Error retrieving contact');
+          reject();
+        });
+      });
+    }
+
+    // TODO: Implement handler for open Vcards
+
+    // Get action from URL (new or update)
+    var params = getParams();
+    if (params.contact) {
+      renderContact(params.contact).then(function(savedContact, count) {
+        DetailsController.setContact(params.contact);
+        DetailsUI.render(savedContact, count, false);
+        window.addEventListener(
+          'pageshow',
```

Comment:
In general I would like to have this listener, event, function name, and a separate function for better reading